edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/DiagnosticConfig.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+    using Microsoft.Extensions.Configuration;
+
+    public class DiagnosticConfig
+    {
+        public readonly bool Enabled;
+        public readonly string MetricsStoragePath;
+        public readonly TimeSpan ScrapeInterval;
+        public readonly TimeSpan UploadInterval;
+
+        public DiagnosticConfig(bool enabled, string storagePath, IConfiguration configuration)
+        {
+            this.Enabled = enabled;
+            this.MetricsStoragePath = Path.Combine(storagePath, "metrics");
+            this.ScrapeInterval = configuration.GetValue("MetricScrapeInterval", TimeSpan.FromHours(1));
```

Comment:
can scrapeInterval be longer than uploadInterval?

Reply:
It can, it just means that some uploads would be empty. I'll add a check to not send a message on an empty upload.